osu.Game/Rulesets/UI/ModIcon.cs

```diff
@@ -123,40 +123,12 @@ private void updateMod(IMod value)
                 modAcronym.FadeOut();
             }
 
-            switch (value.Type)
-            {
-                default:
-                case ModType.DifficultyIncrease:
-                    backgroundColour = colours.Red1;
-                    highlightedColour = colours.Red0;
-                    break;
-
-                case ModType.DifficultyReduction:
-                    backgroundColour = colours.Lime1;
-                    highlightedColour = colours.Lime0;
-                    break;
-
-                case ModType.Automation:
-                    backgroundColour = colours.Blue1;
-                    highlightedColour = colours.Blue0;
-                    break;
-
-                case ModType.Conversion:
-                    backgroundColour = colours.Purple1;
-                    highlightedColour = colours.Purple0;
-                    break;
-
-                case ModType.Fun:
-                    backgroundColour = colours.Pink1;
-                    highlightedColour = colours.Pink0;
-                    break;
-
-                case ModType.System:
-                    backgroundColour = colours.Gray7;
-                    highlightedColour = colours.Gray8;
-                    modIcon.Colour = colours.Yellow;
-                    break;
-            }
+            Color4 typeColour = colours.ForModType(value.Type);
+            backgroundColour = typeColour;
+            highlightedColour = ((Colour4)typeColour).Lighten(.2f);
+
+            if (value.Type == ModType.System)
+                modIcon.Colour = colours.Yellow;
```

Comment:
I do wonder if this can be moved to the `ForModType` return?

@bdach do you recall why it's specified as `Gray` there and if that ever shows up?

https://github.com/ppy/osu/blob/f8830c6850128456266c82de83273204f8b74ac0/osu.Game/Graphics/OsuColour.cs#L186-L187

Reply:
I've removed the yellow local colour logic for now. Using grey feels fine for the time being.